docker/Dockerfile

```diff
@@ -72,11 +72,9 @@ ENV CXX="/usr/lib/ccache/g++"
 # cspell: ignore libcu libnv
 # Set up development environment
 RUN --mount=type=ssh \
-  ./setup-dev-env.sh -y --module all ${SETUP_ARGS} --no-cuda-drivers openadkit \
+  ./setup-dev-env.sh -y --module all ${SETUP_ARGS} --download-artifacts --no-cuda-drivers openadkit \
```

Comment:
I think the change of https://github.com/autowarefoundation/autoware/pull/4771 was reverted. Do you have some reason?

Reply:
We need to decrease the image size to re-enable the `Upload Artifacts` step.